src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -645,6 +652,20 @@ theorem getElem?_setWidth (m : Nat) (x : BitVec n) (i : Nat) :
     getLsbD (setWidth m x) i = (decide (i < m) && getLsbD x i) := by
   simp [getLsbD, toNat_setWidth, Nat.testBit_mod_two_pow]
 
+theorem getMsbD_setWidth {m : Nat} {x : BitVec n} {i : Nat} :
+    getMsbD (setWidth m x) i = (decide (m - n ≤ i) && getMsbD x (i + n - m)) := by
```

Comment:
No objection on this PR, but increasingly I am skeptical that `decide inequality && boolean` is really what we want on the RHS of all these lemmas. It probably should just be `if inequality then boolean else false`. Without a mixed linear arithmetic + SAT solver, this is doing us no favours, and the `if` style makes it much easier to use `split` to get the inequalities in usable form (e.g. for `omega`).

Something to consider; it would be a big refactor.